packages/design-system-scripts/gulp/build.js

```diff
@@ -67,6 +70,32 @@ async function copyAll(dir) {
   return Promise.all(copyTasks);
 }
 
+/**
+ * Used to generate typescript definition files for the core
+ */
+async function generateTypeDefinitionsFromPropTypes(dir) {
+  const src = path.join(dir, 'src', 'components');
+  const srcGlob = getSrcGlob(src).concat([`!${src}/**/index.js`]);
+
+  return streamPromise(
+    gulp
+      .src(srcGlob, { base: src })
+      .pipe(
+        through.obj((file, enc, cb) => {
+          const definition = react2dts.generateFromFile(file.basename, file.path);
+          file.contents = Buffer.from(definition);
+          file.extname = '.d.ts';
+          cb(null, file);
+        })
+      )
+      .pipe(concat('index.d.ts'))
```

Comment:
We simply combine all the definitions into a single file so its easy to include it in a child ds